x-pack/plugins/maps/public/classes/layers/layer.tsx

```diff
@@ -73,6 +73,8 @@ export interface ILayer {
   isLayerLoading(): boolean;
   isLoadingBounds(): boolean;
   isFilteredByGlobalTime(): Promise<boolean>;
+  isEditable(): Promise<boolean>;
```

Comment:
Would it be possible to move these methods to IVectorLayer interface since only vector layers will support editing? The ILayer interface is already so cluttered and these methods really do not belong on the interface.

Then, check if method exists and just cast to IVectorInterface when needed.